src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -204,5 +208,35 @@ protected Task RaiseDocumentActiveContextChangedEventAsync(SourceTextContainer s
                 return Task.CompletedTask;
             }
         }
+
+        private EventMap.EventHandlerSet<EventHandler<T>> GetEventHandlers<T>(string eventName) where T : EventArgs
+        {
+            // this will register features that want to listen to workspace events
+            // lazily first time workspace event is actually fired
+            EnsureWorkspaceEventListeners();
+
+            return _eventMap.GetEventHandlers<EventHandler<T>>(eventName);
+
+            // local function
+            void EnsureWorkspaceEventListeners()
```

Comment:
Should this assert the lock is being held so we don't end up running this twice in parallel?

Reply:
removed